frame/contracts/src/benchmarking.rs

```diff
@@ -0,0 +1,273 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2020 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Benchmarks for the contracts pallet
+
+#![cfg(feature = "runtime-benchmarks")]
+
+use crate::*;
+use crate::Module as Contracts;
+
+use frame_benchmarking::{benchmarks, account};
+use frame_system::{Module as System, RawOrigin};
+use parity_wasm::elements::FuncBody;
+use sp_runtime::traits::Hash;
+
+const SEED: u32 = 0;
+
+macro_rules! load_module {
+    ($name:expr) => {{
+        let code = include_bytes!(concat!("../fixtures/benchmarks/", $name, ".wat"));
+        compile_module::<T>(code)
+    }};
+}
+
+fn compile_module<T: Trait>(code: &[u8]) -> (Vec<u8>, <T::Hashing as Hash>::Output) {
+    let code = sp_std::str::from_utf8(code).expect("Invalid utf8 in wat file.");
+    let binary = wat::parse_str(code).expect("Failed to compile wat file.");
+    let hash = T::Hashing::hash(&binary);
+    (binary, hash)
+}
+
+fn funding<T: Trait>() -> BalanceOf<T> {
+    T::Currency::minimum_balance() * 10_000.into()
+}
+
+fn create_funded_user<T: Trait>(string: &'static str, n: u32) -> T::AccountId {
+	let user = account(string, n, SEED);
+	T::Currency::make_free_balance_be(&user, funding::<T>());
+	user
+}
+
+fn contract_with_call_body<T: Trait>(body: FuncBody) -> (Vec<u8>, <T::Hashing as Hash>::Output) {
+    use parity_wasm::elements::{
+        Instructions, Instruction::End,
+    };
+    let contract = parity_wasm::builder::ModuleBuilder::new()
+        // deploy function (idx 0)
+        .function()
+            .signature().with_params(vec![]).with_return_type(None).build()
+            .body().with_instructions(Instructions::new(vec![End])).build()
+            .build()
+        // call function (idx 1)
+        .function()
+            .signature().with_params(vec![]).with_return_type(None).build()
+            .with_body(body)
+            .build()
+        .export().field("deploy").internal().func(0).build()
+        .export().field("call").internal().func(1).build()
+        .build();
+    let bytes = contract.to_bytes().unwrap();
+    let hash = T::Hashing::hash(&bytes);
+    (bytes, hash)
+}
+
+fn expanded_contract<T: Trait>(expansions: u32) -> (Vec<u8>, <T::Hashing as Hash>::Output) {
```

Comment:
is this size represented in bytes?

Reply:
Yeah you are right I need to rename that argument.